src/app/clusters/service-area-server/service-area-server.cpp

```diff
@@ -0,0 +1,1104 @@
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+
+CHIP_ERROR Instance::ReadSupportedMaps(AttributeValueEncoder & aEncoder)
+{
+    if (mDelegate->GetNumberOfSupportedMaps() == 0)
+    {
+        return aEncoder.EncodeNull();
+    }
+
+    return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR {
+        uint32_t mapIndex = 0;
+        MapStructureWrapper supportedMap;
+
+        while (mDelegate->GetSupportedMapByIndex(mapIndex++, supportedMap))
+        {
+            ReturnErrorOnFailure(encoder.Encode(supportedMap));
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+
+CHIP_ERROR Instance::ReadSelectedLocations(AttributeValueEncoder & aEncoder)
+{
+    if (mDelegate->GetNumberOfSelectedLocations() == 0)
+    {
+        return aEncoder.EncodeNull();
+    }
+
+    return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR {
+        uint32_t locationIndex = 0;
+        uint32_t selectedLocation;
+
+        while (mDelegate->GetSelectedLocationByIndex(locationIndex++, selectedLocation))
+        {
+            ReturnErrorOnFailure(encoder.Encode(selectedLocation));
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+
+CHIP_ERROR Instance::ReadProgress(AttributeValueEncoder & aEncoder)
+{
+    if (mDelegate->GetNumberOfProgressElements() == 0)
+    {
+        return aEncoder.EncodeNull();
+    }
+
+    return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR {
+        uint32_t locationIndex = 0;
+        Structs::ProgressStruct::Type progressElement;
+
+        while (mDelegate->GetProgressElementByIndex(locationIndex++, progressElement))
+        {
+            ReturnErrorOnFailure(encoder.Encode(progressElement));
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+
+//*************************************************************************
+// command handlers
+
+void Instance::HandleSelectLocationsCmd(HandlerContext & ctx, const Commands::SelectLocations::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "Service Area: HandleSelectLocationsCmd");
+
+    uint32_t listIndex = 0;
+    uint32_t oldSelectedLocation;
+    size_t numberOfLocations = 0;
+    bool matchesCurrentSelectedLocations;
+
+    // On receipt of this command the device SHALL respond with a SelectLocationsResponse command.
+    auto exitResponse = [ctx](SelectLocationsStatus status, CharSpan statusText) {
+        Commands::SelectLocationsResponse::Type response{
+            .status     = status,
+            .statusText = Optional(statusText),
+        };
+        ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+    };
+
+    // Get the number of Selected Locations in the command parameter and check that it is valid.
+    if (!req.newLocations.IsNull())
+    {
+        if (CHIP_NO_ERROR != req.newLocations.Value().ComputeSize(&numberOfLocations))
+        {
+            exitResponse(SelectLocationsStatus::kInvalidSet, "decoding of newLocations parameter size failed"_span);
```

Comment:
Per spec this should be INVALID_COMMAND, no?  See "8.8.2.3. Incoming Invoke Request Action" under "Invoke Execution" where it says "incoming data cannot be mapped to the expected data type for a field".

Reply:
Fixed [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/c8358f1e8f8d38af9dbf71135f96d99406791858).